pyicloud/base.py

```diff
@@ -106,8 +135,8 @@ def request(self, method, url, **kwargs):  # pylint: disable=arguments-differ
             if not code and data.get("serverErrorCode"):
                 code = data.get("serverErrorCode")
 
-                if reason:
-                    self._raise_error(code, reason)
+            if reason:
```

Comment:
@Quentame I believe this was an existing bug I found while working on this PR, btw.